compiler/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -1463,10 +1465,30 @@ impl<'a> Resolver<'a> {
             }
             ExpressionKind::Call(call_expr) => {
                 // Get the span and name of path for error reporting
-                let func = self.resolve_expression(*call_expr.func);
+                let func = self.resolve_expression(*call_expr.clone().func);
 
-                let arguments = vecmap(call_expr.arguments, |arg| self.resolve_expression(arg));
+                let arguments =
+                    vecmap(call_expr.clone().arguments, |arg| self.resolve_expression(arg));
                 let location = Location::new(expr.span, self.file);
+
+                if let ExpressionKind::Variable(path) = call_expr.clone().func.kind {
+                    let (hir_ident, _var_scope_index) = self.get_ident_from_path(path);
+                    if hir_ident.id != DefinitionId::dummy_id() {
+                        if let DefinitionKind::Function(func_def_id) =
+                            self.interner.definition(hir_ident.id).kind
+                        {
+                            if let Some(func_meta) = self.interner.func_meta.get(&func_def_id) {
+                                self.interner.add_reference(
+                                    (DependencyId::Function(func_def_id), func_meta.location),
+                                    (
+                                        DependencyId::CallExpression(func),
+                                        Location::new(call_expr.func.span, self.file),
```

Comment:
Instead of adding references during a call, it will be simpler to add them when the function variable is referenced in the `ExpressionKind::Variable` case

Reply:
ty e685934: chore: move collection point